src/jit/codegenxarch.cpp

```diff
@@ -185,16 +185,15 @@ void                CodeGen::genEmitGSCookieCheck(bool pushReg)
         // Handle multi-reg return type values
         if (compiler->compMethodReturnsMultiRegRetType())
         {
-#ifdef FEATURE_UNIX_AMD64_STRUCT_PASSING
             ReturnTypeDesc retTypeDesc;
             retTypeDesc.Initialize(compiler, compiler->info.compMethodInfo->args.retTypeClass);
+            unsigned regCount = retTypeDesc.GetReturnRegCount();
+            assert(regCount == MAX_RET_REG_COUNT);
```

Comment:
Since this is on xarch, mult-reg return means either x86 or x64 unix and in both the cases this code path is hit only when regCount == MAX_RET_REG_COUNT.   I will add a comment to clarify this.